devolutions-gateway/src/api/network_scan.rs

```diff
@@ -0,0 +1,123 @@
+use std::net::IpAddr;
+
+use axum::{
+    extract::{ws::Message, State, WebSocketUpgrade},
+    response::Response,
+};
+
+use network_scanner::scanner::NetworkScannerParams;
+use serde::Serialize;
+
+use crate::{http::HttpError, DgwState};
+
+pub async fn handler(
+    _: State<DgwState>,
```

Comment:
question: Do we absolutely need to have this parameter? Seems unrequired, but maybe axum needs it for compilation.

Reply:
updated, we don't need it, thanks!